### PR TITLE
Jm event copylink

### PR DESCRIPTION
**Pull request by Jean-LucMartel**
Added a new page after clicking the create event button which takes you to the new page with a unique url with a button that copies the url link to clipboard which can be used to share the event link.

### CODE REVIEW

File: src/app/api/events/route.ts [hunk 1]
```diff
@@ -9,17 +9,38 @@ export async function GET(request: Request) {
 
   const url = new URL(request.url)
   const creatorId = url.searchParams.get('creatorId')
+  const eventId = url.searchParams.get('eventId')
 
-  const { data, error } = await supabase
-    .from('events')
-    .select()
-    .eq('creator', creatorId)
+  console.log(creatorId)
```
Comment (pselvana) [inline src/app/api/events/route.ts]: Please remove these log lines.

File: src/app/api/events/route.ts [hunk 2]
```diff
@@ -9,17 +9,38 @@ export async function GET(request: Request) {
...
+  console.log(eventId)
 
-  if (error) {
-    return NextResponse.json({
-      status: 400,
-      message: 'Error fetching events',
-    })
+  if (creatorId !== null) {
+    console.log('in creatorId')
```
Comment (pselvana) [inline src/app/api/events/route.ts]: More.

File: src/app/api/events/route.ts [hunk 3]
```diff
@@ -9,17 +9,38 @@ export async function GET(request: Request) {
...
+
+    const { data, error } = await supabase
+      .from('events')
+      .select()
+      .eq('creator', creatorId)
+
+    if (error) {
+      return NextResponse.json({
```
Comment (pselvana) [inline src/app/api/events/route.ts:26]: Please do a follow up PR but let's get a visible error page that explains the issue. Like "unknown user" here and "unknown event" below.

Comment (pselvana) [reply src/app/api/events/route.ts:26]: Not in the API but as a separate page.

File: src/app/api/events/routeOld2.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { NextResponse } from 'next/server'
```
Comment (pselvana) [inline src/app/api/events/routeOld2.ts]: What is this file? Is this needed?

File: src/app/create-event/page.tsx [hunk 5]
```diff
@@ -43,7 +48,7 @@ export default function CreateEvent() {
   }
 
   // Create Event Button function
-  const handleSubmit = () => {
+  const handleSubmit = async () => {
     console.log(`Title: ${title}`)
```
Comment (pselvana) [inline src/app/create-event/page.tsx]: You can remove these debug lines as well.

File: src/app/view-event/page.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        .then((data) => {
+          console.log(data)
+          setEvents(data)
+        })
+        .catch((error) => {
+          console.error('Error:', error.message)
+        })
+    }
+
+    getTheEvent() // filler UUID for now until local storage is setup
```
Comment (pselvana) [inline src/app/view-event/page.tsx]: This comment doesn't fit this call.

File: src/app/view-event/page.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        .then((response) => {
+          if (!response.ok) {
+            return response.json().then((err) => {
+              throw new Error(err.message)
+            })
+          }
+          return response.json()
+        })
+        .then((data) => {
+          console.log(data)
```
Comment (pselvana) [inline src/app/view-event/page.tsx]: debug message.

File: src/app/view-event/page.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+const ViewEvent = () => {
+  const searchParams = useSearchParams()
+
+  const eventId = searchParams.get('eventId')
+  // State to hold the event URL
+  const [eventUrl, setEventUrl] = useState('')
+  const [events, setEvents] = useState<any[]>([])
+
+  useEffect(() => {
+    console.log('Event ID in URL is ' + eventId)
```
Comment (pselvana) [inline src/app/view-event/page.tsx]: more

File: src/app/view-event/page.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+          console.error('Error:', error.message)
+        })
+    }
+
+    getTheEvent() // filler UUID for now until local storage is setup
+  }, [])
+
+  return (
+    <div>
+      {/* <EventView eventId='39ae671a-21f0-40f2-b901-963317c0c974' /> */}
```
Comment (pselvana) [inline src/app/view-event/page.tsx]: more.

File: src/app/view-event/page.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      <EventView />
+      {events.length > 0 &&
```
Comment (pselvana) [inline src/app/view-event/page.tsx]: What do we want to do if events.length == 0? If there's an error or a UUID that's wrong, what happens?  What should happen?

File: supabase/migrations/20240712231748_create_Events_table.sql [hunk 11]
```diff
@@ -9,5 +9,7 @@ location text,
 timezone text,
 mode text,
 config json,
-creator uuid references Users(id)
+creator uuid references Users(id),
+-- new column to to create a link that can be shared
+shareable_link text
```
Comment (pselvana) [inline supabase/migrations/20240712231748_create_Events_table.sql]: is shareable link needed?  EventID was used above

**Review by pselvana — CHANGES_REQUESTED** [review]
A bunch of clean changes requested.

File: src/app/api/events/route.ts (hunk 1 above)
Comment (Jean-LucMartel) [reply src/app/api/events/route.ts]: removed

File: src/app/api/events/route.ts (hunk 2 above)
Comment (Jean-LucMartel) [reply src/app/api/events/route.ts]: removed

File: src/app/api/events/routeOld2.ts (hunk 4 above)
Comment (Jean-LucMartel) [reply src/app/api/events/routeOld2.ts]: Kept this line in because it already came with main.

File: src/app/create-event/page.tsx (hunk 5 above)
Comment (Jean-LucMartel) [reply src/app/create-event/page.tsx]: Removed the line and lines followed with console.log

File: src/app/view-event/page.tsx (hunk 6 above)
Comment (Jean-LucMartel) [reply src/app/view-event/page.tsx]: Removed the comment

File: src/app/view-event/page.tsx (hunk 7 above)
Comment (Jean-LucMartel) [reply src/app/view-event/page.tsx]: Removed

File: src/app/view-event/page.tsx (hunk 8 above)
Comment (Jean-LucMartel) [reply src/app/view-event/page.tsx]: Removed.

File: src/app/view-event/page.tsx (hunk 9 above)
Comment (Jean-LucMartel) [reply src/app/view-event/page.tsx]: Removed.

File: src/app/view-event/page.tsx (hunk 10 above)
Comment (Jean-LucMartel) [reply src/app/view-event/page.tsx]: Added changes to the file.  Added 'No events found' if events.length == 0

File: supabase/migrations/20240712231748_create_Events_table.sql (hunk 11 above)
Comment (Jean-LucMartel) [reply supabase/migrations/20240712231748_create_Events_table.sql]: Removed this.

File: src/app/api/events/routeOld2.ts (hunk 4 above)
Comment (pselvana) [reply src/app/api/events/routeOld2.ts]: Looks like you deleted this.

**Review by pselvana — APPROVED** [review]

**Pull request merged**